dp-cluster-setup-utility.py

```diff
@@ -882,6 +893,143 @@ def streamsmsgmgr_url(self):
   def host_name(self, service_name, component_name):
     return self.ambari.cluster.service(service_name).component(component_name).host_names()[0]
 
+
+class BeaconProxyTopology:
+  TEMPLATE = """<?xml version="1.0" encoding="UTF-8"?>
+  <topology>
+     <uri>{knox_url}/gateway/beacon-proxy</uri>
+     <name>beacon-proxy</name>
+     <timestamp>{timestamp}</timestamp>
+     <generated>false</generated>
+     <gateway>
+        <provider>
+          <role>authentication</role>
+          <name>HadoopAuth</name>
+          <enabled>true</enabled>
+          <param>
+            <name>config.prefix</name>
+            <value>hadoop.auth.config</value>
+          </param>
+          <param>
+            <name>hadoop.auth.config.signature.secret</name>
+            <value>knox-signature-secret</value>
+          </param>
+          <param>
+            <name>hadoop.auth.config.type</name>
+            <value>kerberos</value>
+          </param>
+          <param>
+            <name>hadoop.auth.config.simple.anonymous.allowed</name>
+            <value>false</value>
+          </param>
+          <param>
+            <name>hadoop.auth.config.token.validity</name>
+            <value>1800</value>
+          </param>
+          <param>
+            <name>hadoop.auth.config.cookie.domain</name>
+            <value>HWQE.HORTONWORKS.COM</value>
+          </param>
+          <param>
+            <name>hadoop.auth.config.cookie.path</name>
+            <value>gateway/beacon-proxy</value>
+          </param>
+          <param>
+            <name>hadoop.auth.config.kerberos.principal</name>
+            <value>HTTP/{knox_host}@HWQE.HORTONWORKS.COM</value>
+          </param>
+          <param>
+            <name>hadoop.auth.config.kerberos.keytab</name>
+            <value>/etc/security/keytabs/spnego.service.keytab</value>
+          </param>
+          <param>
+            <name>hadoop.auth.config.kerberos.name.rules</name>
+            <value>DEFAULT</value>
+          </param>
+        </provider>
+     </gateway>
+     {ranger}
+     {atlas_api}
+     {beacon}
+  </topology>"""
+
+  def __init__(self, ambari, role_names, name='beacon-proxy'):
+    self.ambari = ambari
+    self.role_names = role_names
+    self.name = name
+
+  def deploy(self, knox):
+    self.update_knox_service_defs(knox)
+    template = BeaconProxyTopology.TEMPLATE.format(
+      knox_url = str(knox.base_url),
+      knox_host = self.ambari.cluster.knox_host(),
+      timestamp = int(time.time()),
+      ambari_protocol = self.ambari.base_url.protocol(),
```

Comment:
It looks like some if these template variables are not used in the template.

Reply:
Will remove the unused ones.